doc/source/train/getting-started-xgboost.rst

```diff
@@ -0,0 +1,429 @@
+        -    data = config["data"]
+        -    model = config["model"]
+
+        +    data = load_dataset()
+        +    model = load_model()
+             ...
+
+         trainer = ray.train.xgboost.XGBoostTrainer(train_func, train_loop_config=config, ...)
+
+Ray Train automatically sets up the Rabit communicator for XGBoost, which handles the distributed communication between workers.
+
+Report metrics and save checkpoints
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To persist your checkpoints and monitor training progress, add a
+:class:`ray.train.xgboost.RayTrainReportCallback` utility callback to your Trainer:
+
+
+.. code-block:: diff
+
+     import xgboost
+     from ray.train.xgboost import RayTrainReportCallback
+
+     def train_func():
+        ...
+        bst = xgboost.train(
+            ...,
+            callbacks=[
+                RayTrainReportCallback(
+                    metrics={"loss": "eval-logloss"}, frequency=1
+                )
+            ],
+        )
+        ...
+
+
+Reporting metrics and checkpoints to Ray Train enables integration with Ray Tune and :ref:`fault-tolerant training <train-fault-tolerance>`.
+
+Loading data
+------------
+
+When running distributed XGBoost training, you will need each worker to access a different shard of the dataset.
+
+
+.. testcode:: python
+    :skipif: True
+
+    def get_train_dataset(world_rank) -> xgboost.DMatrix:
+        # Define logic to get the DMatrix for each worker
+        ...
+
+    def get_eval_dataset(world_rank) -> xgboost.DMatrix:
+        # Define logic to get the DMatrix for each worker
+        ...
+
+    def train_func():
+        rank = ray.train.get_world_rank()
+        dtrain = get_train_dataset(rank)
+        deval = get_eval_dataset(rank)
+        ...
+
+A common way to do this is to pre-shard the dataset, so each worker reads from a different partition of the dataset. 
+
+For more flexibility, Ray Data provides a solution for sharding the dataset at runtime.
+
+Use Ray Data to shard the dataset
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:ref:`Ray Data <data>` is a distributed data processing library that allows you to easily shard and distribute your data across multiple workers. 
+
+First, load your entire dataset as a Ray Data Dataset.
+
+.. testcode:: python
+    :skipif: True
+
+    train_dataset = ray.data.from_parquet("s3://...")
+    eval_dataset = ray.data.from_parquet("s3://...")
+
+In the training function, you can access the dataset shards for this worker using :meth:`ray.train.get_dataset_shard`. Convert this into the DMatrix.
+
+
+.. testcode:: python
+    :skipif: True
+
+    def get_dmatrix(dataset_name: str) -> xgboost.DMatrix:
+        shard = ray.train.get_dataset_shard(dataset_name)
+        df = shard.materialize().to_pandas()
+        X, y = df.drop("target", axis=1), df["target"]
+        return xgboost.DMatrix(X, label=y)
+
+    def train_func():
+        dtrain = get_dmatrix("train")
+        deval = get_dmatrix("eval")
+        ...
```

Comment:
for future: let's introduce a prepare_dmatrix utility API which can implement the pyarrow version of this to reduce copies.

Reply:
Explain and show a diagram about the dataset being materialized at this point and being stored in the object store. 

Include details about how this is allows the dataset to be spread across multiple nodes' CPU memory.

Also mention the default 30% object store limit and how to modify this with the env variable and how to estimate how many nodes you need.

Reply:
Can I add a TODO for this? This is a bit more extensive and might be helpful to have as part of a more thorough Data + Train for XGBoost/LightGBM user guide.